scripts/staticprefs-analyze.py

```diff
@@ -0,0 +1,197 @@
+        header_local_path = os.path.join(os.path.dirname(yaml_path), header_name)
+
+        p = os.path.join(generated_dir, header_local_path)
+        read_cpp_analysis_one(p, cpp_symbols)
+
+        for name in os.listdir(generated_dir):
+            if name.startswith('__'):
+                p = os.path.join(generated_dir, name, header_local_path)
+                read_cpp_analysis_one(p, cpp_symbols)
+
+        if ns == 'javascript':
+            # Load SpiderMonkey-specific binding.
+            header_local_path = 'js/public/PrefsGenerated.h'
+
+            p = os.path.join(generated_dir, header_local_path)
+            read_cpp_analysis_one(p, cpp_symbols)
+
+            for name in os.listdir(generated_dir):
+                if name.startswith('__'):
+                    p = os.path.join(generated_dir, name, header_local_path)
+                    read_cpp_analysis_one(p, cpp_symbols)
+
+    return cpp_symbols
+
+
+pref_bindings_map = {}
+
+
+def get_cpp_symbols(name, yaml_path, bindings_local_path, analysis_root):
+    '''Get the corresponding binding C++ symbols for the preference name.
+    Returns an empty list if no binding is found.'''
+    ns = name.split('.')[0]
+
+    if ns in pref_bindings_map:
+        cpp_symbols = pref_bindings_map[ns]
+    else:
+        cpp_symbols = read_cpp_analysis(ns, yaml_path, bindings_local_path, analysis_root)
+        pref_bindings_map[ns] = cpp_symbols
+
+    id = mk_id(name)
+
+    if id not in cpp_symbols:
+        return []
+
+    return cpp_symbols[id]
+
+
+def to_loc_with(lineno, colno, name_len):
+    return f'{lineno}:{colno}-{colno + name_len}'
+
+
+def process_file(yaml_path, bindings_local_path, files_root, analysis_root):
+    records = []
+
+    with open(os.path.join(files_root, yaml_path), 'r') as f:
+        lineno = 1
+        for line in f:
+            m = re.match('^(- +name: )(.+)', line.rstrip())
+            if m:
+                prefix = m.group(1)
+                name = m.group(2)
+
+                loc = to_loc_with(lineno, len(prefix), len(name))
+                pretty = f'StaticPrefs {name}'
+                sym = f'PREFS_{mk_id(name)}'
+
+                slots = []
+
+                cpp_syms = get_cpp_symbols(name, yaml_path, bindings_local_path, analysis_root)
+                for cpp_sym in cpp_syms:
+                    slots.append({
+                        'slotKind': 'getter',
+                        'slotLang': 'cpp',
+                        'ownerLang': 'prefs',
+                        'sym': cpp_sym,
+                    })
+
+                records.append({
+                    'loc': loc,
+                    'source': 1,
+                    'syntax': 'def',
+                    'pretty': pretty,
+                    'sym': sym,
+                })
+
+                records.append({
+                    'loc': loc,
+                    'target': 1,
+                    'kind': 'def',
+                    'pretty': pretty,
```

Comment:
I think we want just the `name` here (without the "StaticPrefs " prefix) so that the identifiers table can find the synthetic `PREFS_` symbol.  That should improve the default search results a lot.

(still reviewing this great patch but I thought I should mention this sooner since I expect it will be the only feedback I have ;)